deeprankcore/features/contact.py

```diff
@@ -13,21 +13,28 @@
 
 MAX_COVALENT_DISTANCE = 2.1
 
-def _get_coulomb_potentials(atoms: List[Atom], distances: np.ndarray) -> np.ndarray:
+def _get_coulomb_potentials(atoms: List[Atom], distances: np.ndarray, cutoff_distance: float) -> np.ndarray:
     """ 
         Calculate Coulomb potentials between between all Atoms in atom.
         Warning: there's no distance cutoff here. The radius of influence is assumed to infinite.
             However, the potential tends to 0 at large distance.
     """
 
+    clamped_distances = distances
+    clamped_distances[distances == 0.0] = 3.0
+
     EPSILON0 = 1.0
     COULOMB_CONSTANT = 332.0636
     charges = [atomic_forcefield.get_charge(atom) for atom in atoms]
-    coulomb_potentials = np.expand_dims(charges, axis=1) * np.expand_dims(charges, axis=0) * COULOMB_CONSTANT / (EPSILON0 * distances)
-    return coulomb_potentials
+    coulomb_potentials = np.expand_dims(charges, axis=1) * np.expand_dims(charges, axis=0) * COULOMB_CONSTANT / (EPSILON0 * clamped_distances)
+
+    cutoff_factors = (1.0 - (clamped_distances / cutoff_distance) ** 2) ** 2
```

Comment:
Where did you take this new formula?

Reply:
It was here:
https://github.com/DeepRank/deeprank/blob/3594e05dca15e688c10ef66e0fbc7cd71fd40027/deeprank/features/AtomicFeature.py#L620